src/libsyntax/parse/parser.rs

```diff
@@ -233,6 +233,57 @@ pub struct ModulePathError {
     pub help_msg: String,
 }
 
+pub enum Errors {
+    FileNotFoundForModule { mod_name: String,
+                            default_path: String,
+                            secondary_path: String,
+                            dir_path: String },
+    DuplicatePaths { mod_name: String, default_path: String, secondary_path: String },
+    UselessDocComment,
+    InclusiveRangeWithNoEnd,
```

Comment:
I think `UselessDocComment` and `InclusiveRangeWithNoEnd` would be more appropriate as methods on the `Parser`, since they don't have anything to do with with the first two variants (except that they are errors during parsing).

Then, the enum could be named `ModulePathError` -- clearer and more precise.